src/main/java/org/fluentd/kafka/GroupConsumer.java

```diff
@@ -41,9 +41,8 @@ public GroupConsumer(PropertyConfig config) throws IOException {
             ZkUtils.maybeDeletePath(config.get(PropertyConfig.Constants.KAFKA_ZOOKEEPER_CONNECT.key), "/consumers/" + config.get(PropertyConfig.Constants.KAFKA_GROUP_ID.key));
     }
 
-    public FluentLogger setupFluentdLogger() {
-        final URI uri = config.getFluentdConnect();
-        return FluentLogger.getLogger("", uri.getHost(), uri.getPort());
+    public Fluency setupFluentdLogger() throws IOException {
+        return Fluency.defaultFluency(config.getFluentdConnect());
     }
  
     public void shutdown() {
```

Comment:
In the following lines, after the timeout of `executor.awaitTermination()`, call `executor.shutdownNow()`